### PR TITLE
Add Burgers observers

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

- Update ScalarWave observer for recent changes
- Add observer to Burgers executable

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp [hunk 1]
```diff
@@ -16,7 +16,8 @@
 #include "Evolution/Systems/ScalarWave/Actions.hpp"  // IWYU pragma: keep
 #include "Evolution/Systems/ScalarWave/Equations.hpp"  // IWYU pragma: keep // for UpwindFlux
 #include "Evolution/Systems/ScalarWave/System.hpp"
-#include "IO/Observer/Actions.hpp"            // IWYU pragma: keep
+#include "IO/Observer/Actions.hpp"  // IWYU pragma: keep
```
Comment (fmahebert) [inline src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp:19]: The ValenciaDivClean observer code lives in a file Observe.hpp. I think it'd be cleaner to adopt the same naming here in ScalarWave. Objections?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Observe.hpp [hunk 2]
```diff
@@ -45,10 +45,6 @@ using reduction_datums =
 
 /*!
  * \brief Temporary action for observing volume and reduction data
```
Comment (fmahebert) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Observe.hpp:47]: @nilsdeppe Are these actions still temporary or have they become what we want in the long term?

File: src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp:19]: Agreed, no objection.

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Observe.hpp (hunk 2 above)
Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/ValenciaDivClean/Observe.hpp:47]: Still temporary

Comment (fmahebert) [reply src/Evolution/Systems/GrMhd/ValenciaDivClean/Observe.hpp:47]: Thanks for confirming; I wanted check while I was touching the nearby code already :+1:

File: src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp (hunk 1 above)
Comment (fmahebert) [reply src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp:19]: Okay, I will rename the file ScalarWave/Actions.hpp to ScalarWave/Observe.hpp, and same for Burgers.

**Comment by fmahebert** [comment]
Updated after first exchange with @nilsdeppe

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe did you want to look at this?

**Review by wthrowe — APPROVED** [review]

**Pull request merged**